libs/sdk-ui-geo/src/core/geoChart/GeoChartInner.tsx

```diff
@@ -277,20 +312,41 @@ export class GeoChartInner extends React.PureComponent<IGeoChartInnerProps, IGeo
             geoData,
         };
         const colorLegendValue: string = colorStrategy.getColorByIndex(0);
+        const legendDetails = this.getLegendDetailOptions(position, contentRect);
+        let legendProps: Partial<IGeoChartLegendRendererProps> = {
+            height,
+            locale,
+            position,
+            responsive,
+            isFluidLegend,
+            onItemClick: this.onLegendItemClick,
+            contentRect,
+        };
+
+        if (legendDetails) {
+            legendProps = {
+                ...legendProps,
+                maxRows: legendDetails.maxRows,
+                name: legendDetails.name,
+                renderPopUp: legendDetails.renderPopUp,
+            };
+        }
 
         if (segment && enabledLegendItems.length) {
             return {
                 ...legendProps,
                 ...propsFromData,
                 categoryItems,
                 colorLegendValue,
+                containerId: this.containerId,
```

Comment:
Nitpick: you could probably set the containerId in the original legendProps creation (thus avoiding duplication).

Reply:
Done https://github.com/gooddata/gooddata-ui-sdk/pull/1341/commits/51496ed5026684eb7684beb6baf17a95be5b9c46